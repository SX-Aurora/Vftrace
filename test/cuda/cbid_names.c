#include <stdio.h>

#include <cupti.h>

int main (int argc, char *argv[]) {
   const char *cbid_name;
   for (int cbid = 0; cbid < 405; cbid++) {
      cuptiGetCallbackName (CUPTI_CB_DOMAIN_RUNTIME_API, cbid, &cbid_name);
      fprintf (stdout, "%d: %s\n", cbid, cbid_name);
   }
   return 0;
}
