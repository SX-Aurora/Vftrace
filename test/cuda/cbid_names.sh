#!/bin/bash
set -x
test_name=cbid_names
output_file=${test_name}.out
ref_file=${srcdir}/ref_output/${output_file}

rm -f ${output_file}

./cbid_names > ${output_file} || exit 1

diff ${output_file} ${ref_file}
