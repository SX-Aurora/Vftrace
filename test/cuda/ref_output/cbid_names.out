0: INVALID
1: cudaDriverGetVersion_v3020
2: cudaRuntimeGetVersion_v3020
3: cudaGetDeviceCount_v3020
4: cudaGetDeviceProperties_v3020
5: cudaChooseDevice_v3020
6: cudaGetChannelDesc_v3020
7: cudaCreateChannelDesc_v3020
8: cudaConfigureCall_v3020
9: cudaSetupArgument_v3020
10: cudaGetLastError_v3020
11: cudaPeekAtLastError_v3020
12: cudaGetErrorString_v3020
13: cudaLaunch_v3020
14: cudaFuncSetCacheConfig_v3020
15: cudaFuncGetAttributes_v3020
16: cudaSetDevice_v3020
17: cudaGetDevice_v3020
18: cudaSetValidDevices_v3020
19: cudaSetDeviceFlags_v3020
20: cudaMalloc_v3020
21: cudaMallocPitch_v3020
22: cudaFree_v3020
23: cudaMallocArray_v3020
24: cudaFreeArray_v3020
25: cudaMallocHost_v3020
26: cudaFreeHost_v3020
27: cudaHostAlloc_v3020
28: cudaHostGetDevicePointer_v3020
29: cudaHostGetFlags_v3020
30: cudaMemGetInfo_v3020
31: cudaMemcpy_v3020
32: cudaMemcpy2D_v3020
33: cudaMemcpyToArray_v3020
34: cudaMemcpy2DToArray_v3020
35: cudaMemcpyFromArray_v3020
36: cudaMemcpy2DFromArray_v3020
37: cudaMemcpyArrayToArray_v3020
38: cudaMemcpy2DArrayToArray_v3020
39: cudaMemcpyToSymbol_v3020
40: cudaMemcpyFromSymbol_v3020
41: cudaMemcpyAsync_v3020
42: cudaMemcpyToArrayAsync_v3020
43: cudaMemcpyFromArrayAsync_v3020
44: cudaMemcpy2DAsync_v3020
45: cudaMemcpy2DToArrayAsync_v3020
46: cudaMemcpy2DFromArrayAsync_v3020
47: cudaMemcpyToSymbolAsync_v3020
48: cudaMemcpyFromSymbolAsync_v3020
49: cudaMemset_v3020
50: cudaMemset2D_v3020
51: cudaMemsetAsync_v3020
52: cudaMemset2DAsync_v3020
53: cudaGetSymbolAddress_v3020
54: cudaGetSymbolSize_v3020
55: cudaBindTexture_v3020
56: cudaBindTexture2D_v3020
57: cudaBindTextureToArray_v3020
58: cudaUnbindTexture_v3020
59: cudaGetTextureAlignmentOffset_v3020
60: cudaGetTextureReference_v3020
61: cudaBindSurfaceToArray_v3020
62: cudaGetSurfaceReference_v3020
63: cudaGLSetGLDevice_v3020
64: cudaGLRegisterBufferObject_v3020
65: cudaGLMapBufferObject_v3020
66: cudaGLUnmapBufferObject_v3020
67: cudaGLUnregisterBufferObject_v3020
68: cudaGLSetBufferObjectMapFlags_v3020
69: cudaGLMapBufferObjectAsync_v3020
70: cudaGLUnmapBufferObjectAsync_v3020
71: cudaWGLGetDevice_v3020
72: cudaGraphicsGLRegisterImage_v3020
73: cudaGraphicsGLRegisterBuffer_v3020
74: cudaGraphicsUnregisterResource_v3020
75: cudaGraphicsResourceSetMapFlags_v3020
76: cudaGraphicsMapResources_v3020
77: cudaGraphicsUnmapResources_v3020
78: cudaGraphicsResourceGetMappedPointer_v3020
79: cudaGraphicsSubResourceGetMappedArray_v3020
80: cudaVDPAUGetDevice_v3020
81: cudaVDPAUSetVDPAUDevice_v3020
82: cudaGraphicsVDPAURegisterVideoSurface_v3020
83: cudaGraphicsVDPAURegisterOutputSurface_v3020
84: cudaD3D11GetDevice_v3020
85: cudaD3D11GetDevices_v3020
86: cudaD3D11SetDirect3DDevice_v3020
87: cudaGraphicsD3D11RegisterResource_v3020
88: cudaD3D10GetDevice_v3020
89: cudaD3D10GetDevices_v3020
90: cudaD3D10SetDirect3DDevice_v3020
91: cudaGraphicsD3D10RegisterResource_v3020
92: cudaD3D10RegisterResource_v3020
93: cudaD3D10UnregisterResource_v3020
94: cudaD3D10MapResources_v3020
95: cudaD3D10UnmapResources_v3020
96: cudaD3D10ResourceSetMapFlags_v3020
97: cudaD3D10ResourceGetSurfaceDimensions_v3020
98: cudaD3D10ResourceGetMappedArray_v3020
99: cudaD3D10ResourceGetMappedPointer_v3020
100: cudaD3D10ResourceGetMappedSize_v3020
101: cudaD3D10ResourceGetMappedPitch_v3020
102: cudaD3D9GetDevice_v3020
103: cudaD3D9GetDevices_v3020
104: cudaD3D9SetDirect3DDevice_v3020
105: cudaD3D9GetDirect3DDevice_v3020
106: cudaGraphicsD3D9RegisterResource_v3020
107: cudaD3D9RegisterResource_v3020
108: cudaD3D9UnregisterResource_v3020
109: cudaD3D9MapResources_v3020
110: cudaD3D9UnmapResources_v3020
111: cudaD3D9ResourceSetMapFlags_v3020
112: cudaD3D9ResourceGetSurfaceDimensions_v3020
113: cudaD3D9ResourceGetMappedArray_v3020
114: cudaD3D9ResourceGetMappedPointer_v3020
115: cudaD3D9ResourceGetMappedSize_v3020
116: cudaD3D9ResourceGetMappedPitch_v3020
117: cudaD3D9Begin_v3020
118: cudaD3D9End_v3020
119: cudaD3D9RegisterVertexBuffer_v3020
120: cudaD3D9UnregisterVertexBuffer_v3020
121: cudaD3D9MapVertexBuffer_v3020
122: cudaD3D9UnmapVertexBuffer_v3020
123: cudaThreadExit_v3020
124: cudaSetDoubleForDevice_v3020
125: cudaSetDoubleForHost_v3020
126: cudaThreadSynchronize_v3020
127: cudaThreadGetLimit_v3020
128: cudaThreadSetLimit_v3020
129: cudaStreamCreate_v3020
130: cudaStreamDestroy_v3020
131: cudaStreamSynchronize_v3020
132: cudaStreamQuery_v3020
133: cudaEventCreate_v3020
134: cudaEventCreateWithFlags_v3020
135: cudaEventRecord_v3020
136: cudaEventDestroy_v3020
137: cudaEventSynchronize_v3020
138: cudaEventQuery_v3020
139: cudaEventElapsedTime_v3020
140: cudaMalloc3D_v3020
141: cudaMalloc3DArray_v3020
142: cudaMemset3D_v3020
143: cudaMemset3DAsync_v3020
144: cudaMemcpy3D_v3020
145: cudaMemcpy3DAsync_v3020
146: cudaThreadSetCacheConfig_v3020
147: cudaStreamWaitEvent_v3020
148: cudaD3D11GetDirect3DDevice_v3020
149: cudaD3D10GetDirect3DDevice_v3020
150: cudaThreadGetCacheConfig_v3020
151: cudaPointerGetAttributes_v4000
152: cudaHostRegister_v4000
153: cudaHostUnregister_v4000
154: cudaDeviceCanAccessPeer_v4000
155: cudaDeviceEnablePeerAccess_v4000
156: cudaDeviceDisablePeerAccess_v4000
157: cudaPeerRegister_v4000
158: cudaPeerUnregister_v4000
159: cudaPeerGetDevicePointer_v4000
160: cudaMemcpyPeer_v4000
161: cudaMemcpyPeerAsync_v4000
162: cudaMemcpy3DPeer_v4000
163: cudaMemcpy3DPeerAsync_v4000
164: cudaDeviceReset_v3020
165: cudaDeviceSynchronize_v3020
166: cudaDeviceGetLimit_v3020
167: cudaDeviceSetLimit_v3020
168: cudaDeviceGetCacheConfig_v3020
169: cudaDeviceSetCacheConfig_v3020
170: cudaProfilerInitialize_v4000
171: cudaProfilerStart_v4000
172: cudaProfilerStop_v4000
173: cudaDeviceGetByPCIBusId_v4010
174: cudaDeviceGetPCIBusId_v4010
175: cudaGLGetDevices_v4010
176: cudaIpcGetEventHandle_v4010
177: cudaIpcOpenEventHandle_v4010
178: cudaIpcGetMemHandle_v4010
179: cudaIpcOpenMemHandle_v4010
180: cudaIpcCloseMemHandle_v4010
181: cudaArrayGetInfo_v4010
182: cudaFuncSetSharedMemConfig_v4020
183: cudaDeviceGetSharedMemConfig_v4020
184: cudaDeviceSetSharedMemConfig_v4020
185: cudaCreateTextureObject_v5000
186: cudaDestroyTextureObject_v5000
187: cudaGetTextureObjectResourceDesc_v5000
188: cudaGetTextureObjectTextureDesc_v5000
189: cudaCreateSurfaceObject_v5000
190: cudaDestroySurfaceObject_v5000
191: cudaGetSurfaceObjectResourceDesc_v5000
192: cudaMallocMipmappedArray_v5000
193: cudaGetMipmappedArrayLevel_v5000
194: cudaFreeMipmappedArray_v5000
195: cudaBindTextureToMipmappedArray_v5000
196: cudaGraphicsResourceGetMappedMipmappedArray_v5000
197: cudaStreamAddCallback_v5000
198: cudaStreamCreateWithFlags_v5000
199: cudaGetTextureObjectResourceViewDesc_v5000
200: cudaDeviceGetAttribute_v5000
201: cudaStreamDestroy_v5050
202: cudaStreamCreateWithPriority_v5050
203: cudaStreamGetPriority_v5050
204: cudaStreamGetFlags_v5050
205: cudaDeviceGetStreamPriorityRange_v5050
206: cudaMallocManaged_v6000
207: cudaOccupancyMaxActiveBlocksPerMultiprocessor_v6000
208: cudaStreamAttachMemAsync_v6000
209: cudaGetErrorName_v6050
210: cudaOccupancyMaxActiveBlocksPerMultiprocessor_v6050
211: cudaLaunchKernel_v7000
212: cudaGetDeviceFlags_v7000
213: cudaLaunch_ptsz_v7000
214: cudaLaunchKernel_ptsz_v7000
215: cudaMemcpy_ptds_v7000
216: cudaMemcpy2D_ptds_v7000
217: cudaMemcpyToArray_ptds_v7000
218: cudaMemcpy2DToArray_ptds_v7000
219: cudaMemcpyFromArray_ptds_v7000
220: cudaMemcpy2DFromArray_ptds_v7000
221: cudaMemcpyArrayToArray_ptds_v7000
222: cudaMemcpy2DArrayToArray_ptds_v7000
223: cudaMemcpyToSymbol_ptds_v7000
224: cudaMemcpyFromSymbol_ptds_v7000
225: cudaMemcpyAsync_ptsz_v7000
226: cudaMemcpyToArrayAsync_ptsz_v7000
227: cudaMemcpyFromArrayAsync_ptsz_v7000
228: cudaMemcpy2DAsync_ptsz_v7000
229: cudaMemcpy2DToArrayAsync_ptsz_v7000
230: cudaMemcpy2DFromArrayAsync_ptsz_v7000
231: cudaMemcpyToSymbolAsync_ptsz_v7000
232: cudaMemcpyFromSymbolAsync_ptsz_v7000
233: cudaMemset_ptds_v7000
234: cudaMemset2D_ptds_v7000
235: cudaMemsetAsync_ptsz_v7000
236: cudaMemset2DAsync_ptsz_v7000
237: cudaStreamGetPriority_ptsz_v7000
238: cudaStreamGetFlags_ptsz_v7000
239: cudaStreamSynchronize_ptsz_v7000
240: cudaStreamQuery_ptsz_v7000
241: cudaStreamAttachMemAsync_ptsz_v7000
242: cudaEventRecord_ptsz_v7000
243: cudaMemset3D_ptds_v7000
244: cudaMemset3DAsync_ptsz_v7000
245: cudaMemcpy3D_ptds_v7000
246: cudaMemcpy3DAsync_ptsz_v7000
247: cudaStreamWaitEvent_ptsz_v7000
248: cudaStreamAddCallback_ptsz_v7000
249: cudaMemcpy3DPeer_ptds_v7000
250: cudaMemcpy3DPeerAsync_ptsz_v7000
251: cudaOccupancyMaxActiveBlocksPerMultiprocessorWithFlags_v7000
252: cudaMemPrefetchAsync_v8000
253: cudaMemPrefetchAsync_ptsz_v8000
254: cudaMemAdvise_v8000
255: cudaDeviceGetP2PAttribute_v8000
256: cudaGraphicsEGLRegisterImage_v7000
257: cudaEGLStreamConsumerConnect_v7000
258: cudaEGLStreamConsumerDisconnect_v7000
259: cudaEGLStreamConsumerAcquireFrame_v7000
260: cudaEGLStreamConsumerReleaseFrame_v7000
261: cudaEGLStreamProducerConnect_v7000
262: cudaEGLStreamProducerDisconnect_v7000
263: cudaEGLStreamProducerPresentFrame_v7000
264: cudaEGLStreamProducerReturnFrame_v7000
265: cudaGraphicsResourceGetMappedEglFrame_v7000
266: cudaMemRangeGetAttribute_v8000
267: cudaMemRangeGetAttributes_v8000
268: cudaEGLStreamConsumerConnectWithFlags_v7000
269: cudaLaunchCooperativeKernel_v9000
270: cudaLaunchCooperativeKernel_ptsz_v9000
271: cudaEventCreateFromEGLSync_v9000
272: cudaLaunchCooperativeKernelMultiDevice_v9000
273: cudaFuncSetAttribute_v9000
274: cudaImportExternalMemory_v10000
275: cudaExternalMemoryGetMappedBuffer_v10000
276: cudaExternalMemoryGetMappedMipmappedArray_v10000
277: cudaDestroyExternalMemory_v10000
278: cudaImportExternalSemaphore_v10000
279: cudaSignalExternalSemaphoresAsync_v10000
280: cudaSignalExternalSemaphoresAsync_ptsz_v10000
281: cudaWaitExternalSemaphoresAsync_v10000
282: cudaWaitExternalSemaphoresAsync_ptsz_v10000
283: cudaDestroyExternalSemaphore_v10000
284: cudaLaunchHostFunc_v10000
285: cudaLaunchHostFunc_ptsz_v10000
286: cudaGraphCreate_v10000
287: cudaGraphKernelNodeGetParams_v10000
288: cudaGraphKernelNodeSetParams_v10000
289: cudaGraphAddKernelNode_v10000
290: cudaGraphAddMemcpyNode_v10000
291: cudaGraphMemcpyNodeGetParams_v10000
292: cudaGraphMemcpyNodeSetParams_v10000
293: cudaGraphAddMemsetNode_v10000
294: cudaGraphMemsetNodeGetParams_v10000
295: cudaGraphMemsetNodeSetParams_v10000
296: cudaGraphAddHostNode_v10000
297: cudaGraphHostNodeGetParams_v10000
298: cudaGraphAddChildGraphNode_v10000
299: cudaGraphChildGraphNodeGetGraph_v10000
300: cudaGraphAddEmptyNode_v10000
301: cudaGraphClone_v10000
302: cudaGraphNodeFindInClone_v10000
303: cudaGraphNodeGetType_v10000
304: cudaGraphGetRootNodes_v10000
305: cudaGraphNodeGetDependencies_v10000
306: cudaGraphNodeGetDependentNodes_v10000
307: cudaGraphAddDependencies_v10000
308: cudaGraphRemoveDependencies_v10000
309: cudaGraphDestroyNode_v10000
310: cudaGraphInstantiate_v10000
311: cudaGraphLaunch_v10000
312: cudaGraphLaunch_ptsz_v10000
313: cudaGraphExecDestroy_v10000
314: cudaGraphDestroy_v10000
315: cudaStreamBeginCapture_v10000
316: cudaStreamBeginCapture_ptsz_v10000
317: cudaStreamIsCapturing_v10000
318: cudaStreamIsCapturing_ptsz_v10000
319: cudaStreamEndCapture_v10000
320: cudaStreamEndCapture_ptsz_v10000
321: cudaGraphHostNodeSetParams_v10000
322: cudaGraphGetNodes_v10000
323: cudaGraphGetEdges_v10000
324: cudaStreamGetCaptureInfo_v10010
325: cudaStreamGetCaptureInfo_ptsz_v10010
326: cudaGraphExecKernelNodeSetParams_v10010
327: cudaThreadExchangeStreamCaptureMode_v10010
328: cudaDeviceGetNvSciSyncAttributes_v10020
329: cudaOccupancyAvailableDynamicSMemPerBlock_v10200
330: cudaStreamSetFlags_v10200
331: cudaStreamSetFlags_ptsz_v10200
332: cudaGraphExecMemcpyNodeSetParams_v10020
333: cudaGraphExecMemsetNodeSetParams_v10020
334: cudaGraphExecHostNodeSetParams_v10020
335: cudaGraphExecUpdate_v10020
336: cudaGetFuncBySymbol_v11000
337: cudaCtxResetPersistingL2Cache_v11000
338: cudaGraphKernelNodeCopyAttributes_v11000
339: cudaGraphKernelNodeGetAttribute_v11000
340: cudaGraphKernelNodeSetAttribute_v11000
341: cudaStreamCopyAttributes_v11000
342: cudaStreamCopyAttributes_ptsz_v11000
343: cudaStreamGetAttribute_v11000
344: cudaStreamGetAttribute_ptsz_v11000
345: cudaStreamSetAttribute_v11000
346: cudaStreamSetAttribute_ptsz_v11000
347: cudaDeviceGetTexture1DLinearMaxWidth_v11010
348: cudaGraphUpload_v10000
349: cudaGraphUpload_ptsz_v10000
350: cudaGraphAddMemcpyNodeToSymbol_v11010
351: cudaGraphAddMemcpyNodeFromSymbol_v11010
352: cudaGraphAddMemcpyNode1D_v11010
353: cudaGraphMemcpyNodeSetParamsToSymbol_v11010
354: cudaGraphMemcpyNodeSetParamsFromSymbol_v11010
355: cudaGraphMemcpyNodeSetParams1D_v11010
356: cudaGraphExecMemcpyNodeSetParamsToSymbol_v11010
357: cudaGraphExecMemcpyNodeSetParamsFromSymbol_v11010
358: cudaGraphExecMemcpyNodeSetParams1D_v11010
359: cudaArrayGetSparseProperties_v11010
360: cudaMipmappedArrayGetSparseProperties_v11010
361: cudaGraphExecChildGraphNodeSetParams_v11010
362: cudaGraphAddEventRecordNode_v11010
363: cudaGraphEventRecordNodeGetEvent_v11010
364: cudaGraphEventRecordNodeSetEvent_v11010
365: cudaGraphAddEventWaitNode_v11010
366: cudaGraphEventWaitNodeGetEvent_v11010
367: cudaGraphEventWaitNodeSetEvent_v11010
368: cudaGraphExecEventRecordNodeSetEvent_v11010
369: cudaGraphExecEventWaitNodeSetEvent_v11010
370: cudaEventRecordWithFlags_v11010
371: cudaEventRecordWithFlags_ptsz_v11010
372: cudaDeviceGetDefaultMemPool_v11020
373: cudaMallocAsync_v11020
374: cudaMallocAsync_ptsz_v11020
375: cudaFreeAsync_v11020
376: cudaFreeAsync_ptsz_v11020
377: cudaMemPoolTrimTo_v11020
378: cudaMemPoolSetAttribute_v11020
379: cudaMemPoolGetAttribute_v11020
380: cudaMemPoolSetAccess_v11020
381: cudaArrayGetPlane_v11020
382: cudaMemPoolGetAccess_v11020
383: cudaMemPoolCreate_v11020
384: cudaMemPoolDestroy_v11020
385: cudaDeviceSetMemPool_v11020
386: cudaDeviceGetMemPool_v11020
387: cudaMemPoolExportToShareableHandle_v11020
388: cudaMemPoolImportFromShareableHandle_v11020
389: cudaMemPoolExportPointer_v11020
390: cudaMemPoolImportPointer_v11020
391: cudaMallocFromPoolAsync_v11020
392: cudaMallocFromPoolAsync_ptsz_v11020
393: cudaSignalExternalSemaphoresAsync_v2_v11020
394: cudaSignalExternalSemaphoresAsync_v2_ptsz_v11020
395: cudaWaitExternalSemaphoresAsync_v2_v11020
396: cudaWaitExternalSemaphoresAsync_v2_ptsz_v11020
397: cudaGraphAddExternalSemaphoresSignalNode_v11020
398: cudaGraphExternalSemaphoresSignalNodeGetParams_v11020
399: cudaGraphExternalSemaphoresSignalNodeSetParams_v11020
400: cudaGraphAddExternalSemaphoresWaitNode_v11020
401: cudaGraphExternalSemaphoresWaitNodeGetParams_v11020
402: cudaGraphExternalSemaphoresWaitNodeSetParams_v11020
403: cudaGraphExecExternalSemaphoresSignalNodeSetParams_v11020
404: cudaGraphExecExternalSemaphoresWaitNodeSetParams_v11020
